src/cryptenroll/cryptenroll.c

```diff
@@ -534,17 +536,31 @@ static int parse_argv(int argc, char *argv[]) {
                 }
         }
 
-        if (optind >= argc)
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                       "No block device node specified, refusing.");
-
         if (argc > optind+1)
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                        "Too many arguments, refusing.");
 
-        r = parse_path_argument(argv[optind], false, &arg_node);
-        if (r < 0)
-                return r;
+        if (optind < argc) {
+                r = parse_path_argument(argv[optind], false, &arg_node);
+                if (r < 0)
+                        return r;
+        } else {
+                dev_t devno;
+
+                r = blockdev_get_root(LOG_ERR, &devno);
+                if (r < 0)
+                        return r;
+                if (r == 0) {
+                        log_error("Root file system not backed by a (single) whole block device.");
+                        return 80; /* some recognizable error code */
```

Comment:
do you actually care about the error?

Note that you have to switch from DEFINE_MAIN_FUNCTION() to DEFINE_MAIN_FUNCTION_WITH_POSITIVE_FAILURE() in the last line of the file if this shall be propagated.

but frankly, i'd just do `return log_error_errno(SYNTHETIC_ERRNO(ENXIO), "…");` here, i.e. return a regular negative error which will then turned into EXIT_FAILURE as last step. Unless you actually really care about a recognizable error.

If you really do want the recognizable error then please try to use a BSD error, i.e. the stuff from `/usr/include/sysexits.h`, maybe EX_PROTOCOL or EX_UNAVAILABLE or so? or maybe even EX_DATAERR...

but again, i'd just not bother

Reply:
ah, right. Too much c&p from bootctl :see_no_evil: